File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/utils/DateRangeDTO.java
package fpt.aptech.server.dto.utils;

import fpt.aptech.server.enums.date.DateRangeType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * DTO chứa thông tin về một khoảng thời gian để hiển thị trên thanh trượt.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DateRangeDTO {
    /**
     * Nhãn hiển thị cho người dùng.
     * VD: "Tháng 5/2026", "Hôm nay", "Q2 2026"
     */
    private String label;

    /**
     * Thời điểm bắt đầu của khoảng thời gian.
     */
    private LocalDateTime startDate;

    /**
     * Thời điểm kết thúc của khoảng thời gian.
     */
    private LocalDateTime endDate;

    /**
     * Phân loại khoảng thời gian này là quá khứ, hiện tại hay tương lai.
     */
    private DateRangeType type;
}

File: APP MOBILE/smart_money/lib/core/models/date_range_dto.dart
import '../enums/date_range_type.dart';

/// Khoảng thời gian cho thanh trượt (tuần/tháng/quý/năm).
/// Tương ứng: DateRangeDTO.java (server)
class DateRangeDTO {
  final String label;
  final DateTime startDate;
  final DateTime endDate;
  final DateRangeType type;

  const DateRangeDTO({
    required this.label,
    required this.startDate,
    required this.endDate,
    required this.type,
  });

  factory DateRangeDTO.fromJson(Map<String, dynamic> json) {
    return DateRangeDTO(
      label: json['label'] as String,
      startDate: DateTime.parse(json['startDate'] as String),
      endDate: DateTime.parse(json['endDate'] as String),
      type: DateRangeType.fromString(json['type'] as String),
    );
  }
}

